lib/elixir/lib/record.ex

```diff
@@ -309,7 +311,7 @@ defmodule Record do
 
   # Callback invoked from record/0 and record/1 macros.
   @doc false
-  def __access__(atom, fields, args, caller) do
+  def __access__(atom, fields, args, caller) when is_atom(atom) and is_map(caller) do
```

Comment:
Those guards don't make much sense because the args should be validated when the calls are defined. Please remove them.